File: app/src/androidTest/java/com/inelasticcollision/recipelink/data/local/migration/helper/TestVersion2SQLiteDataHelper.java
package com.inelasticcollision.recipelink.data.local.migration.helper;

import android.content.ContentValues;

import com.inelasticcollision.recipelink.DataFactory;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

public class TestVersion2SQLiteDataHelper {

    private static final Comparator<TestVersion2Recipe> TITLE_COMPARATOR = new Comparator<TestVersion2Recipe>() {
        @Override
        public int compare(TestVersion2Recipe o1, TestVersion2Recipe o2) {
            return o1.getTitle().compareTo(o2.getTitle());
        }
    };

    public static List<TestVersion2Recipe> generateRandomContentSortedByTitle(int size) {
        ArrayList<TestVersion2Recipe> values = new ArrayList<>(size);
        for (int i = 0; i < size; i++) {
            ContentValues value = new ContentValues(9);
            value.put("date_added", DataFactory.randomLong());
            value.put("date_updated", DataFactory.randomLong());
            value.put("title", DataFactory.randomString());
            value.put("link", DataFactory.randomString());
            value.put("photo_url", DataFactory.randomBoolean() ? DataFactory.randomString() : "");
            value.put("favorite", DataFactory.randomBoolean() ? 1 : 0);
            value.put("category", DataFactory.randomBoolean() ? DataFactory.randomString() : "");
            value.put("keywords", DataFactory.randomBoolean() ? DataFactory.randomString() : "");
            value.put("notes", DataFactory.randomBoolean() ? DataFactory.randomString() : "");
            values.add(new TestVersion2Recipe(value));
        }
        Collections.sort(values, TITLE_COMPARATOR);
        return values;
    }

}

File: app/src/androidTest/java/com/inelasticcollision/recipelink/data/cache/database/helper/TestVersion2SQLiteDataHelper.kt
package com.inelasticcollision.recipelink.data.cache.database.helper

import android.content.ContentValues
import com.inelasticcollision.recipelink.data.model.Recipe
import com.inelasticcollision.recipelink.test.factory.DataFactory.randomBoolean
import com.inelasticcollision.recipelink.test.factory.DataFactory.randomLong
import com.inelasticcollision.recipelink.test.factory.DataFactory.randomString
import com.inelasticcollision.recipelink.util.nullIfEmpty
import java.util.*

object TestVersion2SQLiteDataHelper {

    @JvmStatic
    fun generateRandomContent(size: Int): List<ContentValues> {
        val values = ArrayList<ContentValues>(size)
        for (i in 0 until size) {
            val value = ContentValues(9)
            value.put("date_added", randomLong())
            value.put("date_updated", randomLong())
            value.put("title", randomString())
            value.put("link", randomString())
            value.put("photo_url", if (randomBoolean()) randomString() else "")
            value.put("favorite", if (randomBoolean()) 1 else 0)
            value.put("category", if (randomBoolean()) randomString() else "")
            value.put("keywords", if (randomBoolean()) randomString() else "")
            value.put("notes", if (randomBoolean()) randomString() else "")
            values.add(value)
        }
        return values
    }

    @JvmStatic
    fun mapContentValuesToExpectedRecipes(contentValues: List<ContentValues>): List<Recipe> {
        return contentValues.map { values ->
            return@map Recipe(
                lastModified = Date(values.getAsLong("date_updated")),
                title = values.getAsString("title"),
                url = values.getAsString("link"),
                imageUrl = values.getAsString("photo_url").nullIfEmpty(),
                favorite = values.getAsBoolean("favorite"),
                notes = values.getAsString("notes").nullIfEmpty(),
                tags = values.getAsString("keywords").nullIfEmpty()?.split("|")
            )
        }.sortedBy { recipe ->
            recipe.title
        }
    }
}